src/action-ros-ci.ts

```diff
@@ -629,8 +646,17 @@ done`;
 	await execShellCommand(["vcs log -l1 src/"], options);
 
 	if (isLinux) {
-		// Always update APT before installing packages on Ubuntu
-		await execShellCommand(["sudo apt-get update"]);
+		// Always update package index before installing packages
+		const dist: string = await determineDistrib();
+		if (dist === "ubuntu") {
+			await execShellCommand(["apt update || sudo apt update || true"]);
+		} else if (dist === "almalinux" || dist === "rocky") {
+			await execShellCommand([
+				"dnf check-update || sudo dnf check-update || true",
+			]);
```

Comment:
we would previously fail if the command didn't work (with `sudo`), but now we don't fail if the command doesn't work (with or without `sudo`). I don't think we should hide this potential failure.

Reply:
Are you specifically commenting on the `dnf` branch or also the `apt` branch above?

The `dnf check-update` command works differently from the `apt update` command. It returns with a non-zero exit code when updates are available. I don't think the whole action should fail just because there are updates available. This behaviour did not exist for the `apt` path before and I don't see why it should be added here. This would also be outside of the scope of the PR.

But for the `apt` path above, I guess the `true` should not be there. I probably added it there from a copy&paste error.